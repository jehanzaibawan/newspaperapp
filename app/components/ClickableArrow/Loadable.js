/**
 *
 * Asynchronously loads the component for ClickableArrow
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
