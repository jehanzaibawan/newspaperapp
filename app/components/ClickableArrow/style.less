.npApp_clickable-arrow {
  span {
    display: inline-flex;
    align-self: center;
    cursor: pointer;

    &:before {
      font-family: 'all-icon-font';
      font-size: 1.2rem;
      content: '\E90C';
      color: #727272;
    }
  }
}