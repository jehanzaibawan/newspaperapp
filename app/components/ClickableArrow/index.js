/**
 *
 * ClickableArrow
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import './style.less';

export const keyDownEventHandler = () => true;

export const ClickableArrow = props => (
  <div className={`npApp_clickable-arrow ${props.className}`}>
    <span
      data-testid="list-item-arrow"
      onClick={props.onClick}
      role="button"
      tabIndex={0}
      onKeyDown={keyDownEventHandler}
    />
  </div>
);

ClickableArrow.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

ClickableArrow.defaultProps = {
  className: '',
};

export default memo(ClickableArrow);
