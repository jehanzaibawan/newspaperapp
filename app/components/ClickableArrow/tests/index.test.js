/**
 *
 * Tests for DateWithIcon
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ClickableArrow, { keyDownEventHandler } from '../index';

describe('<ClickableArrow />', () => {
  it('Should render and match the snapshot', () => {
    const props = {
      onClick: () => {},
    };

    const {
      container: { firstChild },
    } = render(<ClickableArrow {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});

describe('keyDownEventHandler()', () => {
  it('Should be call when key down event is triggered', () => {
    expect(keyDownEventHandler()).toBe(true);
  });
});
