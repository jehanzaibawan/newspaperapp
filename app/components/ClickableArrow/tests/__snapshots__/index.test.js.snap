// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClickableArrow /> Should render and match the snapshot 1`] = `
<div
  class="npApp_clickable-arrow "
>
  <span
    data-testid="list-item-arrow"
    role="button"
    tabindex="0"
  />
</div>
`;
